Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -242,27 +267,29 @@ def Print(s, file=sys.stdout):
     file.write(s + '\n')
 
 # Compare resource version in DSC and agent machine
-# Returns 
+# Returns
 #   0 if version is same
 #   dsc version number if there is a mismatch or agent config not present
 def TestResourceVersion():
     retval = 0
     dscVersion = ReadFile(DSC_RESOURCE_VERSION_PATH)
-    if not os.path.exists(AGENT_RESOURCE_VERSION_PATH):
+    agentVersionFile = os.path.join(PATH_NPM_STATE, FNAME_NPM_VERSION)
+    if not os.path.exists(agentVersionFile):
         #npmd agent is not present, copy binaries
         retval = dscVersion
     else:
-        agentVersion = ReadFile(AGENT_RESOURCE_VERSION_PATH)
-        if agentVersion != dscVersion:
-            #version mismatch, copy binaries
+        agentVersion = ReadFile(agentVersionFile)
+        if LooseVersion(agentVersion) < LooseVersion(dscVersion):
+            #version is new, copy binaries
```

Comment:
The previous check seems more appropriate, the key aspect is to maintain dsc version consistent with agent version irrespective of whether it is older or newer.